thrift/src/test/java/com/linecorp/armeria/it/metrics/PrometheusMetricsIntegrationTest.java

```diff
@@ -0,0 +1,312 @@
+
+        final RequestContext ctx = log.context();
+        final Object requestEnvelope = log.requestEnvelope();
+        final Object requestContent = log.requestContent();
+
+        String pathAsMetricName = null;
+        String methodName = null;
+
+        if (requestEnvelope instanceof HttpHeaders) {
+            pathAsMetricName = ctx.path();
+            methodName = ((HttpHeaders) requestEnvelope).method().name();
+        }
+
+        if (requestContent instanceof RpcRequest) {
+            methodName = ((RpcRequest) requestContent).method();
+        }
+
+        pathAsMetricName = MoreObjects.firstNonNull(pathAsMetricName, "__UNKNOWN_PATH__");
+
+        if (methodName == null) {
+            methodName = MoreObjects.firstNonNull(log.method(), "__UNKNOWN_METHOD__");
+        }
+
+        return new MyMetricLabel(metricNamePrefix,
+                                 "Armeria client/server request metric",
+                                 new MyMetricValue(serviceName, methodName, pathAsMetricName));
+    }
+
+    /**
+     * Returns the name of the service metric.
+     */
+    public static String[] serviceMetricName() {
+        return new String[] { "armeria", "server" };
+    }
+
+    /**
+     * Returns the name of the client metric.
+     */
+    public static String[] clientMetricName() {
+        return new String[] { "armeria", "client" };
+    }
+
+    public static class MyMetricValue {
+        private String handler;
+        private String method;
+        private String path;
+
+        public MyMetricValue(String handler, String method, String path) {
+            this.handler = handler;
+            this.method = method;
+            this.path = path;
+        }
+
+        public String getHandler() {
+            return handler;
+        }
+
+        public String getMethod() {
+            return method;
+        }
+
+        public String getPath() {
+            return path;
+        }
+    }
+
+    public static class MyMetricLabel implements MetricLabel<MyMetricValue> {
+
+        private String[] metricNamePrefix;
+        private String description;
+        private MyMetricValue myMetricValue;
+
+        public MyMetricLabel(String[] metricNamePrefix, String description, MyMetricValue myMetricValue) {
+            this.metricNamePrefix = metricNamePrefix;
+            this.description = description;
+            this.myMetricValue = myMetricValue;
+        }
+
+        @Override
+        public List<String> getTokenizedName() {
+            return ImmutableList.<String>builder().add(metricNamePrefix).build();
+        }
+
+        @Override
+        public String getDescription() {
+            return description;
+        }
+
+        @Override
+        public MyMetricValue getValue() {
+            return myMetricValue;
+        }
+    }
+
+    @Test(timeout = 10000L)
+    public void normal() throws Exception {
+        latch = new CountDownLatch(14);
```

Comment:
I think you can use Awaitlity instead of CountDownLatch to wait request complete